examples/vision/depth_estimation.py

```diff
@@ -0,0 +1,488 @@
+
+        image_ = cv2.imread(image_path)
+        image_ = cv2.cvtColor(image_, cv2.COLOR_BGR2RGB)
+        image_ = cv2.resize(image_, self.dim)
+        image_ = tf.image.convert_image_dtype(image_, tf.float32)
+
+        depth_map = np.load(depth_map).squeeze()
+
+        mask = np.load(mask)
+        mask = mask > 0
+
+        max_depth = min(300, np.percentile(depth_map, 99))
+        depth_map = np.clip(depth_map, self.min_depth, max_depth)
+        depth_map = np.log(depth_map, where=mask)
+
+        depth_map = np.ma.masked_where(~mask, depth_map)
+
+        depth_map = np.clip(depth_map, 0.1, np.log(max_depth))
+        depth_map = cv2.resize(depth_map, self.dim)
+        depth_map = np.expand_dims(depth_map, axis=2)
+        depth_map = tf.image.convert_image_dtype(depth_map, tf.float32)
+
+        return image_, depth_map
+
+    def data_generation(self, batch):
+        x = np.empty((self.batch_size, *self.dim, self.n_channels))
+        y = np.empty((self.batch_size, *self.dim, 1))
+
+        for i, batch_id in enumerate(batch):
+            x[i,], y[i,] = self.load(
+                self.data["image"][batch_id],
+                self.data["depth"][batch_id],
+                self.data["mask"][batch_id],
+            )
+        return x, y
+
+
+"""
+## Visualizing samples
+"""
+
+
+def visualize_depth_map(samples, test=False, model=None):
+    input, target = samples
+    cmap = plt.cm.jet
+    cmap.set_bad(color="black")
+
+    if test:
+        pred = model.predict(input)
+        fig, ax = plt.subplots(6, 3, figsize=(50, 50))
+        for i in range(6):
+            ax[i, 0].imshow((input[i].squeeze()))
+            ax[i, 1].imshow((target[i].squeeze()), cmap=cmap)
+            ax[i, 2].imshow((pred[i].squeeze()), cmap=cmap)
+
+    else:
+        fig, ax = plt.subplots(6, 2, figsize=(50, 50))
+        for i in range(6):
+            ax[i, 0].imshow((input[i].squeeze()))
+            ax[i, 1].imshow((target[i].squeeze()), cmap=cmap)
+
+
+visualize_samples = next(
+    iter(DataGenerator(data=df, batch_size=6, dim=(HEIGHT, WIDTH)))
+)
+visualize_depth_map(visualize_samples)
+
+"""
+## 3D point cloud visualization
+"""
+
+depth_vis = np.flipud(visualize_samples[1][1].squeeze())  # target
+img_vis = np.flipud(visualize_samples[0][1].squeeze())  # input
+
+fig = plt.figure(figsize=(15, 10))
+ax = plt.axes(projection="3d")
+
+STEP = 3
+for x in range(0, img_vis.shape[0], STEP):
+    for y in range(0, img_vis.shape[1], STEP):
+        ax.scatter(
+            [depth_vis[x, y]] * 3,
+            [y] * 3,
+            [x] * 3,
+            c=tuple(img_vis[x, y, :3] / 255),
+            s=3,
+        )
+    ax.view_init(45, 135)
+
+"""
+## Building the model
+
+1. The basic model is a U-Net.
+2. Addditive skip-connections are implemented in the downscaling block.
+"""
+
+
+class DownscaleBlock(layers.Layer):
+    def __init__(self, filters, kernel_size=(3, 3), padding="same", strides=1):
+        super().__init__()
```

Comment:
In all custom layers you define, add `**kwargs` in the signature and pass them to `super().__init__(**kwargs)`